sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts

```diff
@@ -248,47 +248,46 @@ export class ProgramInfoRequestController extends BaseController {
         throw new BadRequestException("Application not found.");
       }
 
+      let studyStartDate = payload.studyStartDate;
+      let studyEndDate = payload.studyEndDate;
       let selectedOfferingIntensity = payload.offeringIntensity;
 
-      if (payload.selectedOffering) {
-        const offering = await this.offeringService.getOfferingById(
-          payload.selectedOffering,
-        );
-        selectedOfferingIntensity = offering.offeringIntensity;
-      }
-
-      this.applicationService.checkOfferingIntensityMismatch(
-        application.data.howWillYouBeAttendingTheProgram,
-        selectedOfferingIntensity,
-      );
-
       let offeringToCompletePIR: EducationProgramOffering;
       if (payload.selectedOffering) {
         // Check if the offering belongs to the location.
-        const offeringLocationId =
+        const offeringLocation =
           await this.offeringService.getOfferingLocationId(
             payload.selectedOffering,
           );
-        if (offeringLocationId !== locationId) {
+
+        if (offeringLocation.institutionLocation.id !== locationId) {
```

Comment:
I believe that there is a missing check to verify if the _offeringLocation_ exists.
For now maybe we can just mark the _offeringLocation_ as possible nullable as below:
```
if (offeringLocation?.institutionLocation.id !== locationId)
```